dark/reads.py

```diff
@@ -133,6 +135,26 @@ def walkHSP(self, hsp):
             readOffset += 1
             subjectOffset += 1
 
+    def checkAlphabeth(self, count=10):
+        """
+        A function which checks whether the sequence in a L{dark.Read} object
+        corresponds to its readClass. For AA reads, more testing is done in
+        dark.Read.AARead.checkAlphabeth.
+
+        @param count: A C{int} of how many amino acids should be considered.
+        @return: C{True} if the sequence corresponds from its readClass or
+            C{False} if the sequence doesn't correspond.
+        @raise ValueError: If the sequence and the readClass don't match, raise
+            a ValueError.
+        """
+        readLetters = set(self.sequence.upper()[:count])
+        # Check if readLetters is a subset of self.ALPHABETH.
+        if self.ALPHABETH is None or readLetters.issubset(self.ALPHABETH):
+            return readLetters
+        raise ValueError("It seems like you're trying to make a Read object "
```

Comment:
The read object is already made, so the error isn't really accurate.  I'd be more specific and provide more information:

```python
raise ValueError("Read alphabet (%r) is not a subset of expected alphabet (%r) for read class %s" % (
    readLetters, self.ALPHABET, self.__class__.__name__))
```